### ISSUE TITLE
caf::result stops automatic flattening of std::tuple

### ISSUE THREAD

**Issue by alatdneg** [actor-framework/actor-framework#1041]
Not sure if this is intended, or I'm just using it wrong.

[=](do_a_thing_atom) -> caf::result<std::tuple<int,bool>>  (we use response_promise here)

[=](do_b_thing_atom) -> std::tuple<int,bool> 

When I receive from do_a_thing_atom, the tuple is not flattened, where as do_b_thing_atom it is.

I'd assume the use of caf::result to be transparent to the caller. Is this a bug ? 

cheers

**Comment by Neverlord** [comment]
> Is this a bug ?

Yes, there's no reason why the API should break the symmetry here.

**Comment by Neverlord** [comment]
I think we should restore symmetry by *never* flattening `std::tuple`. See https://github.com/actor-framework/actor-framework/issues/1048.
